protocol/x/vault/keeper/orders.go

```diff
@@ -30,10 +32,79 @@ const (
 )
 
 // RefreshAllVaultOrders refreshes all orders for all vaults by
-// TODO(TRA-134)
 // 1. Cancelling all existing orders.
 // 2. Placing new orders.
 func (k Keeper) RefreshAllVaultOrders(ctx sdk.Context) {
+	// Iterate through all vaults.
+	totalSharesIterator := k.getTotalSharesIterator(ctx)
+	defer totalSharesIterator.Close()
+	for ; totalSharesIterator.Valid(); totalSharesIterator.Next() {
+		var vaultId types.VaultId
+		k.cdc.MustUnmarshal(totalSharesIterator.Key(), &vaultId)
+		var totalShares types.NumShares
+		k.cdc.MustUnmarshal(totalSharesIterator.Value(), &totalShares)
+
+		// Skip if TotalShares is non-positive.
+		if totalShares.NumShares.Cmp(dtypes.NewInt(0)) <= 0 {
+			continue
+		}
+
+		// Refresh orders depending on vault type.
+		switch vaultId.Type {
+		case types.VaultType_VAULT_TYPE_CLOB:
+			err := k.RefreshVaultClobOrders(ctx, vaultId)
+			if err != nil {
+				log.ErrorLogWithError(ctx, "Failed to refresh vault clob orders", err, "vaultId", vaultId)
+			}
+		}
+	}
+}
+
+// RefreshVaultClobOrders refreshes orders of a CLOB vault.
+func (k Keeper) RefreshVaultClobOrders(ctx sdk.Context, vaultId types.VaultId) (err error) {
+	// Cancel CLOB orders from last block.
+	ordersToCancel, err := k.GetVaultClobOrders(
+		ctx.WithBlockHeight(ctx.BlockHeight()-1),
+		vaultId,
+	)
+	if err != nil {
+		log.ErrorLogWithError(ctx, "Failed to get vault clob orders to cancel", err, "vaultId", vaultId)
+		return err
+	}
+	for _, order := range ordersToCancel {
+		if _, exists := k.clobKeeper.GetLongTermOrderPlacement(ctx, order.OrderId); exists {
+			err := k.clobKeeper.HandleMsgCancelOrder(ctx, clobtypes.NewMsgCancelOrderStateful(
+				order.OrderId,
+				uint32(ctx.BlockTime().Unix())+ORDER_EXPIRATION_SECONDS,
+			))
+			if err != nil {
+				log.ErrorLogWithError(ctx, "Failed to cancel order", err, "order", order, "vaultId", vaultId)
+			}
+			vaultId.IncrCounterWithLabels(
+				metrics.VaultCancelOrder,
+				metrics.GetLabelForBoolValue(metrics.Success, err == nil),
+			)
+		}
+	}
+
+	// Place new CLOB orders.
+	ordersToPlace, err := k.GetVaultClobOrders(ctx, vaultId)
+	if err != nil {
+		log.ErrorLogWithError(ctx, "Failed to get vault clob orders to place", err, "vaultId", vaultId)
+		return err
+	}
+	for _, order := range ordersToPlace {
+		err := k.clobKeeper.HandleMsgPlaceOrder(ctx, clobtypes.NewMsgPlaceOrder(*order))
```

Comment:
Should we make the optimization if the order already exists, do not remove it? Would save us two operations

Reply:
every block, we want to refresh these orders so that their sizes and prices are updated

Reply:
yep there can be. created a ticket [here](https://linear.app/dydx/issue/TRA-169/look-for-ways-to-minimize-order-placement-cancellation) to work on optimization later